src/providers/azure.js

```diff
@@ -0,0 +1,192 @@
+/*
+ * Copyright 2019 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+/* eslint-disable no-param-reassign */
+
+'use strict';
+
+const pick = require('lodash.pick');
+const rp = require('request-promise-native');
+
+const mapResult = {
+  created: (path, update) => ({
+    status: 201,
+    path,
+    update,
+  }),
+  moved: (path, oldLocation, update) => ({
+    status: 301,
+    path,
+    movedFrom: oldLocation,
+    update,
+  }),
+  notFound: (attributes, gone) => {
+    const [name] = Object.keys(pick(attributes, ['path', 'sourceHash']));
```

Comment:
I don't understand this. you create a new object with `path` and `sourcHash`, get the keys (`path` and `sourceHash` and choose the first one as `name`... but the order of the keys is not deterministic.....

Reply:
OK, that is indeed not clear: `notFound` is only ever called with **either** `path` or `sourceHash` never with both. So I take the one that has been provided, and add a message with either a `path` or `sourceHash` entry.

Reply:
Indeed: `pick` creates a new object, and adds the found properties in order they appear in the list `['a','b']`: shouldn't that guarantee, that the first key with index **0** is either `a` (if present) or `b` otherwise?

Reply:
According to [this article](https://2ality.com/2015/10/property-traversal-order-es6.html), it should work:
>First, the keys that are integer indices (what these are is explained later), in ascending numeric order.
> Then, all other string keys, in the order in which they were added to the object.

Reply:
I never trust the object key order....

also it's a bit overkill, isn't it?

```
const name = 'path' in attributes ? 'path' : 'sourceHash';
```